src/api/helpers/create-an-application/index.ts

```diff
@@ -1,152 +1,78 @@
 
     if (!account) {
+      console.info('Rejecting application creation - no account found (createAnApplication helper)');
+
       return null;
     }
 
-    /**
-     * Get a country's ID from the provided ISO code.
-     * This is required to be used in:
-     * 1) Eligibility buyer country relationship
-     * 2) Buyer country relationship
-     */
-    const { buyerCountryIsoCode, totalContractValueId, coverPeriodId, ...otherEligibilityAnswers } = eligibilityAnswers;
-
-    const country = await getCountryByField(context, 'isoCode', buyerCountryIsoCode);
-
-    // add default submission type
-    const submissionType = SUBMISSION_TYPE.MIA;
-
     /**
      * Create the initial application.
-     * We need to do this first so that we can use the application ID.
+     * This needs to be done first so that we can use the application ID.
      */
-    const application = await context.db.Application.createOne({
-      data: {
-        owner: {
-          connect: { id: accountId },
-        },
-        status,
-        submissionType,
-      },
+    const application = await initialApplication.create({
+      context,
+      accountId,
+      status,
     });
 
     const { id: applicationId } = application;
 
-    /**
-     * 1) Create a new buyer with country and application relationship.
-     * 2) Get a totalContractValue DB entry, for linking a relationship to eligibility.
-     * 3) Create a cover period value from the DB.
-     * 4) Create a new eligibility with country and application relationship.
-     * 5) Create a new export contract with application relationship.
-     * 6) Create a new policy with application relationship.
-     * 7) Create a new nominated loss payee with application relationship.
-     * 8) Create a new company with application relationship.
-     * 9) Create a new sectionReview with application relationship
-     */
-    const { buyer } = await createABuyer(context, country.id, applicationId);
-
-    const totalContractValue = await getTotalContractValueByField(context, 'valueId', totalContractValueId);
-
-    const coverPeriod = await getCreditPeriodValueByField(context, 'valueId', coverPeriodId);
-
-    const eligibility = await createAnEligibility(context, country.id, applicationId, coverPeriod.id, totalContractValue.id, otherEligibilityAnswers);
-
-    const { exportContract } = await createAnExportContract(context, applicationId);
-
-    const { policy } = await createAPolicy(context, applicationId);
-
-    const nominatedLossPayee = await createANominatedLossPayee(context, applicationId);
-
-    const company = await createACompany(context, applicationId, companyData);
-
-    const sectionReview = await createASectionReview(context, applicationId, sectionReviewData);
+    // create application relationships
+    const { buyerId, companyId, eligibilityId, exportContractId, nominatedLossPayeeId, policyId, sectionReviewId } = await applicationRelationships.create({
```

Comment:
is this the cleanest or is it better to do `const relationships` here? and then `relationships.buyerId` on line 62

Reply:
I think what's in place (destructuring) is cleaner - otherwise it would pollute the readability on line 62-68 since the params would not be destructured?